x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/EsqlFunctionRegistry.java

```diff
@@ -413,8 +414,9 @@ private static FunctionDefinition[][] functions() {
                 def(MvSum.class, MvSum::new, "mv_sum"),
                 def(Split.class, Split::new, "split") },
             // fulltext functions
-            new FunctionDefinition[] { def(Match.class, Match::new, "match"), def(QueryString.class, QueryString::new, "qstr") } };
-
+            new FunctionDefinition[] { def(Match.class, Match::new, "match"), def(QueryString.class, QueryString::new, "qstr") },
+            // hash
+            new FunctionDefinition[] { def(Hash.class, Hash::new, "hash") } };
```

Comment:
I think it's just a regular old string function. I think.

I'd just stick it next to LTRIM

Reply:
Add also short hands for common algorithms as noted in #98545 - SHA_0, SHA_1 plus SHAKE_128.
For that you'd simply extend the main hash function and use a built in algorithm:
```
public Md5 extends Hash {
       public Sha0(Expression input) {
              super("md5", input)
       }
}
```

It's a small thing that we'll be appreciated by the security folks

Reply:
Makes sense. If nobody minds I would like to add those aliases in a separate followup pr to keep this one smaller.

Re supported algorithms, I believe the set of supported ones could be different for different JVMs.
In openjdk javadocs I am seeing:
```
 * <p> Every implementation of the Java platform is required to support
 * the following standard {@code MessageDigest} algorithms:
 * <ul>
 * <li>{@code SHA-1}</li>
 * <li>{@code SHA-256}</li>
```
In https://docs.oracle.com/javase/7/docs/api/java/security/MessageDigest.html this list is following:
```
Every implementation of the Java platform is required to support the following standard MessageDigest algorithms:
* MD5
* SHA-1
* SHA-256
```

I have checked list of supported algorithms with `Security.getAlgorithms("MessageDigest")` in various JVMs and was getting the same `[SHA3-512, SHA-1, SHA-384, SHA3-384, SHA-224, SHA-512/256, SHA-256, MD2, SHA-512/224, SHA3-256, SHA-512, SHA3-224, MD5]` set in
* 17.0.2-open
* 21.0.2-open
* 23-open
* 23.0.1-oracle
* 17-jbr
* eclipse adoptium 17

So we could easily add aliases for MD5 and SHA, but SHAKE would require a some custom implementation.